### PR TITLE
docs: torch DataLoader example with caveats

### LINKED ISSUE #416: Zarrs Codec deadlocking with  num_workers>0 iwth Torch.Dataloaders
I added a comment here where we are erroring in the first `HCSDataloader`. This is not a viscy issue, so logging it here first. 
https://github.com/dl-janelia/image_translation/pull/17#issuecomment-4560375173

The deadlock is not zarr-python's asyncio loop. It's zarrs — the Rust codec pipeline.

CC: @alxndrkalinin

### PR DESCRIPTION
**Pull request by srivarra**
Added a quick example/doc with using pytorch's dataloaders.

remaining todos:
- [x] Add issue to `zarrs-python` - https://github.com/zarrs/zarrs-python/issues/171

### CODE REVIEW

File: docs/examples/torch_dataloader.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        # Open lazily and read a single index from the "0" array.
+        with open_ome_zarr(self.store_path, layout="fov", mode="r") as fov:
+            return torch.from_numpy(fov["0"][idx])
+```
+
+## What breaks
+
+!!! danger "Deadlocks on Linux"
+
+    ```python title="train.py" hl_lines="3 7"
```
Comment (edyoshikun) [inline docs/examples/torch_dataloader.md:35]: this doesnt render well in the markdown. take a look at this one and the ones below.

File: docs/examples/torch_dataloader.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    The hang needs **both** a `fork` start method **and** a main-process decode
+    before iterating. A pipeline that reads *only* inside workers may never see it,
+    while the same code that first reads an array in the main process deadlocks
+    every time. That is exactly why this bug is easy to miss in testing.
+
+## How to do it correctly
+
+The simplest fix is to read in the main process, with no workers and no `fork()`:
+
+!!! success "Load in the main process"
```
Comment (edyoshikun) [inline docs/examples/torch_dataloader.md:63]: not sure if we should just replace these with emojis so you dont have to use any CSS

**Review by edyoshikun — COMMENTED** [review]
in the markdown some of the pieces of text dont render well, but I'm not sure if in the website this is fine.

**Comment by edyoshikun** [comment]
Is there a way to look at the zensical website within a PR?

**Comment by srivarra** [comment]
@edyoshikun 

| Is there a way to look at the zensical website within a PR?

Yeah checkout the branch and run `uv run zensical serve --open`.

File: docs/examples/torch_dataloader.md (hunk 1 above)
Comment (srivarra) [reply docs/examples/torch_dataloader.md:35]: these are designed to take advantage of the features in zensical, so they look quite nice once built.

**Review by edyoshikun — APPROVED** [review]
LGTM! thanks
